install_spoolman.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+
+# Check if the script is running under sudo
+if [ "$EUID" -eq 0 ]; then
+    echo "Please do not run this script with sudo. Run it as a regular user."
+    exit 1
+fi
+
+# Get the current user's home directory
+USER_HOME=$(eval echo "~$SUDO_USER")
+
+# Function to check if a user is in a group
+user_in_group() {
+    local user="$1"
+    local group="$2"
+    groups "$user" | grep -q "\b$group\b"
+}
+
+# ANSI escape codes for formatting
+RED='\e[31m'  # Red text
+BLUE='\e[34m'  # Blue text
+UNDERLINE='\e[4m'  # Underlined text
+BOLD='\e[1m'  # Bold text
+RESET='\e[0m' # Reset formatting
+
+# Check if Docker is installed
+if ! command -v docker &>/dev/null; then
+    echo "Docker is not installed. Installing Docker..."
+
+    # Install Docker
+    curl -fsSL https://get.docker.com | sudo sh
+
+    echo "Docker has been installed."
+
+else
+    echo "Docker is already installed."
+fi
+
+# Check if users were added to the group and prompt for a reboot
+if user_in_group "$USER" "docker"; then
+    echo "Users are in the docker group. Continuing with Docker container setup."
+else
+	echo "Adding $SUDO_USER to the docker group..."
+    sudo usermod -aG docker "$SUDO_USER"
+
+	echo "Adding $USER to the docker group..."
+    sudo usermod -aG docker "$USER"
+
+    # Prompt the user for reboot with a warning to save active tasks
+    echo -e "${RED}${BOLD}WARNING: System needs to restart for permissions to take effect. Please make sure to save progress on active tasks before rebooting!${RESET}\n"
+    read -p "Do you want to reboot now? (y/n): " choice
+    if [ "$choice" == "y" ]; then
+        echo "Rebooting the system..."
+        sudo reboot
+        exit
+    elif [ "$choice" == "n" ]; then
+        echo "Reboot the system and then re-run this script to finish the installation."
+        exit
+    else
+        echo "Invalid choice. Please enter 'y' to reboot or 'n' to exit."
+    fi
+fi
+
+# Create the folder and docker-compose.yml file
+INSTALL_PATH="$USER_HOME/printer_data/config/Spoolman"
+mkdir -p "$INSTALL_PATH"
+mkdir -p "$INSTALL_PATH/data"
+cat <<EOF > "$INSTALL_PATH/docker-compose.yml"
+version: '3.8'
+services:
+  spoolman:
+    image: ghcr.io/donkie/spoolman:latest
+    restart: unless-stopped
+    volumes:
+      - type: bind
+        source: "$INSTALL_PATH"
+        target: /home/app/.local/share/spoolman
+    ports:
+      - "7912:8000"
+    environment:
+      - TZ=$(cat /etc/timezone) # Set timezone based on system timezone
+EOF
+
+# Change timezone to match the system timezone
+echo "Changing timezone to match the system timezone..."
+echo $(cat /etc/timezone) | sudo tee /etc/timezone
```

Comment:
thats makes no sense. you read /etc/timezone to write to /etc/timezone? what you do in line 81 is better and more then enough.

Reply:
Yep, I think I mentioned I was using chatgpt to help write the script and help with testing. For some reason it misunderstood that I only wanted to pull the system tz for the container. I had tried to get it to explain why it was there and it said it was needed but I finally got it to understand that I didn't need it there.
I am still learning how to use this programming language and it's been 15yrs since I've really done much programming in general so I'm a bit rusty with what I do know lol. It's removed now.